src/components/transition/CSSTransitionAnimator.js

```diff
@@ -217,6 +219,18 @@ export function createCSSTransitionAnimator(
       pushState(DEFAULT_PARSED_PROPS);
       removeElementStyles(element);
     },
-    finished: () => --remainingPropsToChange <= 0,
+    transitionEnd: (event: TransitionEvent) => {
```

Comment:
@bartoszlorek imho we shouldn't mix event code inside this function. It should be simple 

```
if (--remainingPropsToChange === 0 && finishCallbackRef) {
        finishCallbackRef();
      }
```

the part 
```
 // ignores bubbling events of its own descendants
      if (event.target !== event.currentTarget) {
        return;
      }
```
should be directly. 

Now I understand why called it transitionHandler in the first place, but imho checking `event.target !== event.currentTarget` has nothing to do with animator component. It strongly relies how we add listener to the node (we can e.g stop bubbling with different event)

Reply:
This is not blocker to this PR, but I think we somehow mixing those 2 responsibilities too much.

Reply:
thx!

Reply:
https://github.com/brainly/style-guide/pull/2332/commits/421b9e594fef7679e3475491547be5ef959d82ca